From f92beaf4325cb697827107778aed8579de58418b Mon Sep 17 00:00:00 2001
From: Andrew Kelley <andrew@ziglang.org>
Date: Thu, 21 Aug 2025 18:00:51 -0700
Subject: [PATCH 115/133] compiler: disable InternPool.debug_state as it is
 unsound
X-Developer-Signature: v=1; a=openpgp-sha256; l=831; i=xtex@astrafall.org;
 h=from:subject; bh=2Y2f3O+8vp4zNsq9mulyWUI/SdhQi4a4Ujb7MMjcUxg=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78ng5Nn/cH751UX/tzBjDUod43ubIqd/uvotb+XBq2
 qMHd861dUxkYRDjYrAUU2QpMmzwZtVJ5xddVi4LM4eVCWSItEgDAwMDAwsDX25iXqmRjpGeqbah
 nqGhjpGOEQMXpwBMdd5eRobZp96Eu3z+sY1tgciJpuaOhNW+/xzSO9yD3278k1ZZyzuXkWGuom3
 vxIvht1NeimfmffFp3XyshkdcsP3ps78bXfZ8FWECAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

There can be more than one InternPool instance active at the same time.

(cherry picked from commit 3764f7b0f2b2cac05f7884c93b4da3898c72cd5d)
---
 src/InternPool.zig | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/InternPool.zig b/src/InternPool.zig
index fc6634d92561..0afa7e34f4c1 100644
--- a/src/InternPool.zig
+++ b/src/InternPool.zig
@@ -6940,10 +6940,7 @@ pub fn deactivate(ip: *const InternPool) void {
 
 /// For debugger access only.
 const debug_state = struct {
-    const enable = switch (builtin.zig_backend) {
-        else => false,
-        .stage2_x86_64 => !builtin.strip_debug_info,
-    };
+    const enable = false;
     const enable_checks = enable and !builtin.single_threaded;
     threadlocal var intern_pool: ?*const InternPool = null;
 };
-- 
2.52.0

